### PR TITLE
Data source views everywhere wave 2

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR refactors the recently implemented logic for managing vaults from the settings. We have decided to consistently use a data source view for each data source, meaning that when browsing, this part will now only be described in terms of data source views. Other logic will continue to use the underlying data source when adding or updating data.
It eliminates all `DataSourceOrView` variables and types to streamline everything with `DataSourceView`.

## Risk

Worst case it breaks some logic in the vault settings currently gated behind a FF. Safe to rollback.

### CODE REVIEW

File: front/lib/connector_providers.ts [hunk 1]
```diff
@@ -141,9 +141,10 @@ export const CONNECTOR_CONFIGURATIONS: Record<
   },
 };
 
-export function getDataSourceOrViewName(dsv: DataSourceOrViewInfo) {
+export function getDataSourceNameFromView(dsv: DataSourceViewType): string {
```
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/lib/connector_providers.ts:144]: Oh good catch!

[the remark this answers is not in the document]
Comment (Fraggle) [reply front/lib/connector_providers.ts:144]: (and [this](https://github.com/dust-tt/dust/blob/0d32b5e60e1b0ae92d63af3e515755ab338e8545/front/pages/w/%5BwId%5D/builder/data-sources/%5Bname%5D/index.tsx#L1142) could be refactored to use it)

File: front/pages/api/w/[wId]/vaults/[vId]/data_source_views/index.ts [hunk 2]
```diff
@@ -66,13 +66,22 @@ async function handler(
         vault
       );
 
+      console.log(">> vault:", vault, vault.isSystem());
+      console.log(">> category:", category);
+      console.log(">> dataSourceViews:", dataSourceViews);
```
Comment (Fraggle) [inline front/pages/api/w/[wId]/vaults/[vId]/data_source_views/index.ts:71]: it happens to the best of us then ? :)

**Review by Fraggle — APPROVED** [review]

File: front/pages/api/w/[wId]/vaults/[vId]/data_source_views/index.ts (hunk 2 above)
Comment (flvndvd) [reply front/pages/api/w/[wId]/vaults/[vId]/data_source_views/index.ts:71]: 🙈 😅

File: front/lib/connector_providers.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/lib/connector_providers.ts:144]: Broken link?

**Pull request merged**